lib/netsuite/rest/utilities/roles.rb

```diff
@@ -0,0 +1,85 @@
+require 'net/http'
+require 'uri'
+require 'json'
+
+module NetSuite
+  module Rest
+    module Utilities
+      class Roles
+
+        ROLES_API = "https://rest.netsuite.com/rest/roles"
+
+        def self.get(credentials)
+          new(credentials).get
+        end
+
+        def initialize(credentials)
+          @email     = encode_email(credentials.fetch(:email))
+          @signature = credentials.fetch(:password)
+        end
+
+        def get
+          response = make_request
+          parsed   = JSON.parse(response.body)
+          if reponse.code.to_i == 200
+            extract(parsed)
+          else
+            parsed
+          end
+        end
+
+        private
+
+        def encode_email(unencoded_email)
+          URI.escape(
+            unencoded_email,
+            Regexp.new("[^#{URI::PATTERN::UNRESERVED}]")
+          )
```

Comment:
@iloveitaly 

> why is this required?

You mean the email encoding? I'm not sure why NetSuite requires it. But I couldn't get it to work without it.

Would this be more descriptive?

```ruby
URI.escape(email, /[#{URI::PATTERN::RESERVED}]/)
```